### PR TITLE
CUDA port for AEV computation

### PR DESCRIPTION
**Pull request by akkamesh**
This PR introduces CUDA extension to perform AEV computation efficiently on GPUs. Performance improvement is expected to be ~6X for AEV computation and ~2X for overall training workflow.

### CODE REVIEW

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:494]: I think we should run this on stream `c10::cuda::getCurrentCUDAStream()`, instead of the default stream.

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:516]: I think we should run this on stream `c10::cuda::getCurrentCUDAStream()`, instead of the default stream.

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:531]: I think we should run this on stream `c10::cuda::getCurrentCUDAStream()`, instead of the default stream.

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:587]: I think we should run this on stream `c10::cuda::getCurrentCUDAStream()`, instead of the default stream.

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [reply torchani/extension/aev.cu]: Working at https://github.com/akkamesh/torchani/pull/1

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [reply torchani/extension/aev.cu:474]: Working at https://github.com/akkamesh/torchani/pull/1

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [reply torchani/extension/aev.cu]: Working at https://github.com/akkamesh/torchani/pull/1

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [reply torchani/extension/aev.cu]: Working at https://github.com/akkamesh/torchani/pull/1

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [reply torchani/extension/aev.cu]: Working at https://github.com/akkamesh/torchani/pull/1

**Comment by yueyericardo** [comment]
WIP

- [ ] cuaev correctness test
- [ ] readme for install cuaev

**Review by zasdfgbnm — COMMENTED** [review]
I am seeing the following error
```
============================================================================================================================= test session starts ==============================================================================================================================
platform linux -- Python 3.8.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/gaoxiang/torchani/.hypothesis/examples')
rootdir: /home/gaoxiang/torchani
plugins: hypothesis-5.14.0, xonsh-0.9.22
collected 3 items                                                                                                                                                                                                                                                              

tests/test_cuaev.py::TestCUAEV::testNIST FAILED                                                                                                                                                                                                                          [ 33%]
tests/test_cuaev.py::TestCUAEV::testSimple FAILED                                                                                                                                                                                                                        [ 66%]
tests/test_cuaev.py::TestCUAEV::testTripeptideMD FAILED                                                                                                                                                                                                                  [100%]

=================================================================================================================================== FAILURES ===================================================================================================================================
______________________________________________________________________________________________________________________________ TestCUAEV.testNIST ______________________________________________________________________________________________________________________________

self = <test_cuaev.TestCUAEV testMethod=testNIST>

    def testNIST(self):
        datafile = os.path.join(path, 'test_data/NIST/all')
        with open(datafile, 'rb') as f:
            data = pickle.load(f)
            for coordinates, species, _, _, _, _ in data:
                coordinates = torch.from_numpy(coordinates).to(torch.float).to(self.device)
                species = torch.from_numpy(species).to(self.device)
                _, aev = self.aev_computer((species, coordinates))
                _, cu_aev = self.cuaev_computer((species, coordinates))
                max_diff = (cu_aev - aev).abs().max().item()
>               self.assertLess(max_diff, self.tolerance)
E               AssertionError: 2.641306161880493 not less than 5e-05

tests/test_cuaev.py:69: AssertionError
_____________________________________________________________________________________________________________________________ TestCUAEV.testSimple _____________________________________________________________________________________________________________________________

self = <test_cuaev.TestCUAEV testMethod=testSimple>

    def testSimple(self):
        coordinates = torch.tensor([
            [[0.03192167, 0.00638559, 0.01301679],
             [-0.83140486, 0.39370209, -0.26395324],
             [-0.66518241, -0.84461308, 0.20759389],
             [0.45554739, 0.54289633, 0.81170881],
             [0.66091919, -0.16799635, -0.91037834]],
            [[-4.1862600, 0.0575700, -0.0381200],
             [-3.1689400, 0.0523700, 0.0200000],
             [-4.4978600, 0.8211300, 0.5604100],
             [-4.4978700, -0.8000100, 0.4155600],
             [0.00000000, -0.00000000, -0.00000000]]
        ], requires_grad=True, device=self.device)
        species = torch.tensor([[1, 0, 0, 0, 0], [2, 0, 0, 0, -1]], device=self.device)
    
        _, aev = self.aev_computer((species, coordinates))
        _, cu_aev = self.cuaev_computer((species, coordinates))
        max_diff = (cu_aev - aev).abs().max().item()
>       self.assertLess(max_diff, self.tolerance)
E       AssertionError: 3.756709098815918 not less than 5e-05

tests/test_cuaev.py:45: AssertionError
__________________________________________________________________________________________________________________________ TestCUAEV.testTripeptideMD __________________________________________________________________________________________________________________________

self = <test_cuaev.TestCUAEV testMethod=testTripeptideMD>

    def testTripeptideMD(self):
        for i in range(100):
            datafile = os.path.join(path, 'test_data/tripeptide-md/{}.dat'.format(i))
            with open(datafile, 'rb') as f:
                coordinates, species, _, _, _, _, _, _ = pickle.load(f)
                coordinates = torch.from_numpy(coordinates).float().unsqueeze(0).to(self.device)
                species = torch.from_numpy(species).unsqueeze(0).to(self.device)
                _, aev = self.aev_computer((species, coordinates))
                _, cu_aev = self.cuaev_computer((species, coordinates))
                max_diff = (cu_aev - aev).abs().max().item()
>               self.assertLess(max_diff, self.tolerance)
E               AssertionError: 3.0681800842285156 not less than 5e-05

tests/test_cuaev.py:57: AssertionError
=============================================================================================================================== warnings summary ===============================================================================================================================
tests/test_cuaev.py::TestCUAEV::testNIST
  /home/gaoxiang/torchani/torchani/aev.py:208: UserWarning: This overload of nonzero is deprecated:
        nonzero()
  Consider using one of the following signatures instead:
        nonzero(*, bool as_tuple) (Triggered internally at  ../torch/csrc/utils/python_arg_parser.cpp:962.)
    in_cutoff = (distances <= cutoff).nonzero()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========================================================================================================================= 3 failed, 1 warning in 3.49s =========================================================================================================================
```

**Comment by yueyericardo** [comment]
Looks like it's cuda 11 specific problem. Tests passed on cuda 10/9 gpus.

**Comment by akkamesh** [comment]
> Looks like it's cuda 11 specific problem. Tests passed on cuda 10/9 gpus.

@yueyericardo - I couldn't reproduce this error. Do you still see failures with Cuda 11? Is it only on 2080ti?

**Comment by zasdfgbnm** [comment]
I am now seeing

```
======================================================================
ERROR: testNIST (__main__.TestCUAEV)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_cuaev.py", line 68, in testNIST
    max_diff = (cu_aev - aev).abs().max().item()
RuntimeError: CUDA error: invalid configuration argument
Exception raised from launch_vectorized_kernel at /home/gaoxiang/pytorch-master/aten/src/ATen/native/cuda/CUDALoops.cuh:119 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x5c (0x7f93583d2fac in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: void at::native::gpu_kernel_impl<at::native::AddFunctor<float> >(at::TensorIterator&, at::native::AddFunctor<float> const&) + 0x693 (0x7f93595f5833 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #2: <unknown function> + 0x11c6eeb (0x7f93595b9eeb in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #3: at::native::add_kernel_cuda(at::TensorIterator&, c10::Scalar) + 0x2a (0x7f93595bb41a in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x11c847d (0x7f93595bb47d in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0x11fcb43 (0x7f9385e7fb43 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #6: at::native::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0x9f (0x7f9385e79a0f in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x2701544 (0x7f935aaf4544 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x273e6d4 (0x7f935ab316d4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #9: <unknown function> + 0x17bd5d6 (0x7f93864405d6 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #10: at::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0xf1 (0x7f9386441681 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x2ace4bc (0x7f93877514bc in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x2ace7b4 (0x7f93877517b4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #13: <unknown function> + 0x19c2426 (0x7f9386645426 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #14: at::Tensor::sub(at::Tensor const&, c10::Scalar) const + 0xf1 (0x7f9386645fe1 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x4c2947 (0x7f938bfc9947 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #16: <unknown function> + 0x4c2b77 (0x7f938bfc9b77 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>


======================================================================
ERROR: testSimple (__main__.TestCUAEV)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_cuaev.py", line 44, in testSimple
    max_diff = (cu_aev - aev).abs().max().item()
RuntimeError: CUDA error: invalid configuration argument
Exception raised from launch_vectorized_kernel at /home/gaoxiang/pytorch-master/aten/src/ATen/native/cuda/CUDALoops.cuh:119 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x5c (0x7f93583d2fac in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: void at::native::gpu_kernel_impl<at::native::AddFunctor<float> >(at::TensorIterator&, at::native::AddFunctor<float> const&) + 0x693 (0x7f93595f5833 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #2: <unknown function> + 0x11c6eeb (0x7f93595b9eeb in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #3: at::native::add_kernel_cuda(at::TensorIterator&, c10::Scalar) + 0x2a (0x7f93595bb41a in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x11c847d (0x7f93595bb47d in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0x11fcb43 (0x7f9385e7fb43 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #6: at::native::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0x9f (0x7f9385e79a0f in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x2701544 (0x7f935aaf4544 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x273e6d4 (0x7f935ab316d4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #9: <unknown function> + 0x17bd5d6 (0x7f93864405d6 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #10: at::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0xf1 (0x7f9386441681 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x2ace4bc (0x7f93877514bc in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x2ace7b4 (0x7f93877517b4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #13: <unknown function> + 0x19c2426 (0x7f9386645426 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #14: at::Tensor::sub(at::Tensor const&, c10::Scalar) const + 0xf1 (0x7f9386645fe1 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x4c2947 (0x7f938bfc9947 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #16: <unknown function> + 0x4c2b77 (0x7f938bfc9b77 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>


======================================================================
ERROR: testTripeptideMD (__main__.TestCUAEV)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_cuaev.py", line 56, in testTripeptideMD
    max_diff = (cu_aev - aev).abs().max().item()
RuntimeError: CUDA error: an illegal memory access was encountered
Exception raised from launch_vectorized_kernel at /home/gaoxiang/pytorch-master/aten/src/ATen/native/cuda/CUDALoops.cuh:119 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x5c (0x7f93583d2fac in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libc10.so)
frame #1: void at::native::gpu_kernel_impl<at::native::AddFunctor<float> >(at::TensorIterator&, at::native::AddFunctor<float> const&) + 0x693 (0x7f93595f5833 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #2: <unknown function> + 0x11c6eeb (0x7f93595b9eeb in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #3: at::native::add_kernel_cuda(at::TensorIterator&, c10::Scalar) + 0x2a (0x7f93595bb41a in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #4: <unknown function> + 0x11c847d (0x7f93595bb47d in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #5: <unknown function> + 0x11fcb43 (0x7f9385e7fb43 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #6: at::native::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0x9f (0x7f9385e79a0f in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #7: <unknown function> + 0x2701544 (0x7f935aaf4544 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #8: <unknown function> + 0x273e6d4 (0x7f935ab316d4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cuda.so)
frame #9: <unknown function> + 0x17bd5d6 (0x7f93864405d6 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #10: at::sub(at::Tensor const&, at::Tensor const&, c10::Scalar) + 0xf1 (0x7f9386441681 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #11: <unknown function> + 0x2ace4bc (0x7f93877514bc in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #12: <unknown function> + 0x2ace7b4 (0x7f93877517b4 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #13: <unknown function> + 0x19c2426 (0x7f9386645426 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #14: at::Tensor::sub(at::Tensor const&, c10::Scalar) const + 0xf1 (0x7f9386645fe1 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_cpu.so)
frame #15: <unknown function> + 0x4c2947 (0x7f938bfc9947 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
frame #16: <unknown function> + 0x4c2b77 (0x7f938bfc9b77 in /home/gaoxiang/.local/lib/python3.8/site-packages/torch/lib/libtorch_python.so)
<omitting python frames>


----------------------------------------------------------------------
Ran 3 tests in 2.218s

FAILED (errors=3)
```

It's strange that this kernel is a pytorch kernel, and doing `export CUDA_LAUNCH_BLOCKING=1` does not change this error message. And the same error happens on both 2080ti and 3090.

**Comment by IgnacioJPickering** [comment]
Hello, I'm sorry, this past few days I've been working on some stuff and I haven't had time to review this. What is the current state of the PR?

File: torchani/extension/README.md
Comment (IgnacioJPickering) [inline torchani/extension/README.md:32]: Why are there different benchmarks for shuffle and without shuffle? I don't think shuffling batches should affect the AEVs at all, right?

**Comment by zasdfgbnm** [comment]
@IgnacioJPickering This PR is mostly good. Except that I am seeing errors on CUDA 11.1

Update: Tested on CUDA 10.2, and the error is gone. So seems to be a problem reproducible with 11.1 only. cc: @akkamesh

File: torchani/extension/README.md
Comment (yueyericardo) [reply torchani/extension/README.md:32]: Yes, you are right. There is no big difference now for cuaev. But for python aev, there is still difference. I think it's not bad to keep it here, as it's already tested.

**Comment by yueyericardo** [comment]
Torchscript is important for @IgnacioJPickering 's torchani-amber interface, when I try to regiester cuaev with torchscript, I got some error, which I put at #518 
This PR is good to go, torchscript could be fixed in next PR.

**Comment by akkamesh** [comment]
> @IgnacioJPickering This PR is mostly good. Except that I am seeing errors on CUDA 11.1
> 
> Update: Tested on CUDA 10.2, and the error is gone. So seems to be a problem reproducible with 11.1 only. cc: @akkamesh

Thanks. I can repro this issue now, looking into it.

**Comment by hadim** [comment]
Sorry to jump on this PR. I am the maintainer of the torchani conda forge package at https://github.com/conda-forge/torchani-feedstock and I am trying to anticipate the release of a new torchani version with the new support for Cuda.

Building torchani against multiple versions of cuda should be ok on conda forge but from the standpoint of the build logic that would be simpler if the Cuda extension setup is part of the regular `setup.py` file. You could disable the build of the extension if cuda is not detected for example.

Having a single setup.py file would be helpful.

Then I think we could follow the pytorch convention and have a torchani package for cpu, cuda10, cuda10.1, cuda11, etc that is automatically picked whenever conda detects cudatoolkit or not in the env. The user will still be able to disable the use of Cuda in torchani with the correct option within its code directly if he wants.

Happy to hear your thoughts.

**Comment by zasdfgbnm** [comment]
@hadim What you said makes sense. And it would be great that conda will have torchani with cuaev support out of the box. I will start looking at adding such a build, probably in a separate PR. (What I am likely to do is first add something that builds an empty `.cu` file in the main setup.py, and if that gets merged successfully, I can rebase this PR to use that approach.)

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:1]: Is NVTX used? Should this line be removed?

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:2]: same here, is this include used?

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:4]: Also this one.

File: torchani/extension/aev.cu
Comment (zasdfgbnm) [inline torchani/extension/aev.cu:7]: Also this

**Comment by hadim** [comment]
Sounds good @zasdfgbnm > Let's sync once this PR is merged. In the meantime, I am doing my own build experiment at https://github.com/conda-forge/torchani-feedstock/pull/1 using a fork of this PR: https://github.com/hadim/torchani/blob/cuda/setup.py

**Comment by zasdfgbnm** [comment]
@hadim Sure! Thanks for the work!

**Comment by yueyericardo** [comment]
Current cuaev maybe designed for training purpose, molecule expected to be < 100 atoms.
But as molecule size increase to 1000 atoms or more, cuaev become slower than pyaev.
Benchmark results are in #522

**Comment by zasdfgbnm** [comment]
The idea at https://github.com/aiqm/torchani/pull/516#issuecomment-722073107 is implemented in https://github.com/aiqm/torchani/pull/521, currently this PR is just a draft, it is on early stage and is not working yet.

File: torchani/extension/aev.cu
Comment (akkamesh) [reply torchani/extension/aev.cu]: Removed all unused directives. Marking as resolved for all.

**Comment by akkamesh** [comment]
> > @IgnacioJPickering This PR is mostly good. Except that I am seeing errors on CUDA 11.1
> > Update: Tested on CUDA 10.2, and the error is gone. So seems to be a problem reproducible with 11.1 only. cc: @akkamesh
> 
> Thanks. I can repro this issue now, looking into it.

The issue seems to be outside of ani. I'm trying to repro it with a simple use case without the torchani workflow.

File: torchani/cuaev/setup.py
Comment (hadim) [inline torchani/cuaev/setup.py:40]: It's very minor I guess but can I suggest to move the `cuaev` module into `torchani.cuaev` so it's scoped within `torchani`? The python code using `cuaev` will need to be fixed too.

Comment (zasdfgbnm) [reply torchani/cuaev/setup.py:40]: This `setup.py` will be eventually abandoned. So please ignore this file for now.

Comment (zasdfgbnm) [reply torchani/cuaev/setup.py:40]: In the last commit I am just renaming it, so that it will conflict with https://github.com/aiqm/torchani/pull/521

**Review by yueyericardo — APPROVED** [review]

**Comment by yueyericardo** [comment]
Hey guys, this pr is too long, let's merge this first. #549 cuda 11 error could be fixed at next PR. 
@akkamesh

**Pull request merged**
